lib/publiccloud/azure.pm

```diff
@@ -124,16 +131,61 @@ sub upload_img {
 
     my $key = $self->get_storage_account_keys($storage_account);
 
-    assert_script_run('az storage blob upload --max-connections 4 --account-name '
-          . $storage_account . ' --account-key ' . $key . ' --container-name ' . $self->container
-          . ' --type page --file ' . $file . ' --name ' . $img_name, timeout => 60 * 60 * 2);
-    assert_script_run('az disk create --resource-group ' . $self->resource_group . ' --name ' . $disk_name
-          . ' --source https://' . $storage_account . '.blob.core.windows.net/' . $self->container . '/' . $img_name
-          . ' --hyper-v-generation=' . $sku . ' --architecture=' . $arch);
 
-    assert_script_run('az image create --resource-group ' . $self->resource_group . ' --name ' . $img_name
-          . ' --os-type Linux --hyper-v-generation=' . $sku . ' --source=' . $disk_name);
 
+    # Check if blob already exists
+    my $container = $self->container;
+    my $blobs = script_output("az storage blob list --container-name '$container' --account-name '$storage_account' | jq '.[].name'");
```

Comment:
we already doing this here https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/publiccloud/upload_image.pm#L31 

it is not expected that you getting into this function when image is already there . I am fine to **move** this check into the function but I find it not effective to do this two times

Reply:
`find_img` checks for existing disk images. In the Arm64 case we are not creating disk images, so that check doesn't work there.

Also checking for a already existing blob is useful in all cases, e.g. when the upload might be interrupted.

There are no downsides in doing this check for all images, and it is necessary for Arm64 blob, where we are not creating disk images.

Reply:
> find_img checks for existing disk images. In the Arm64 case we are not creating disk images, so that check doesn't work there.

first of all for Arm64 images currently we running some code which does not work . the fact that you see green bubble does not change this statement . You running code which does not make sense and ALWAYS return you same result. This is bad practice and this is downside. 

> Checking for a already existing blob is anyway useful in all cases, e.g. when the subsequent creation of a disk is interrupted or fails.

in case of  "x86"  `find_img` will return you nothing in such case so again this line will always pass for "x86" case 

what if you will move  this check inside `if ($arch eq 'Arm64') {` and add condition to "find_img" call so we will skip it for "Arm64" ?

Reply:
> First of all for Arm64 images currently we running some code which does not work .

Not true. Can you @asmorodskyi point out where we already upload Arm64 images?

> 
> > Checking for a already existing blob is anyway useful in all cases, e.g. when the subsequent creation of a disk is interrupted or fails.
> 
> in case of "x86" `find_img` will return you nothing in such case so again this line will always pass for "x86" case
> 
> what if you will move this check inside `if ($arch eq 'Arm64') {` and add condition to "find_img" call so we will skip it for "Arm64" ?

And? The line does change anything. It just prints the blocks.

Reply:
> first of all for Arm64 images currently we running some code which does not work . the fact that you see green bubble does not change this statement . You running code which does not make sense and ALWAYS return you same result. This is bad practice and this is downside.

We have a passing VR and I used the uploaded image to create a test VM. I fail to see how this would be possible, if the upload process would not work.

Please point me to the place which doesn't make any sense because I'm completely lost here.

Reply:
ok, resolved